Enforces calling other TypeSpec decorator using `context.call` instead of calling the decorator function directly.

Calling the decorator function directly can result in diagnostics with incorrect location.

## Rule Details

## How to Use

```jsonc
{
  "@typespec/call-decorator": "warn",
}
```

<!--tabs-->

#### ‚ùå Incorrect

```ts
function $foo(context: DecoratorContext, target: Type) {}

function $bar(context: DecoratorContext, target: Type) {
  $foo(context, target);
}
```

```ts
function $foo(context: DecoratorContext, target: Type, name: string) {}

function $bar(context: DecoratorContext, target: Type, name: string) {
  $foo(context, target, `bar.${name}`);
}
```

#### ‚úÖ Correct

```ts
function $foo(context: DecoratorContext, target: Type) {}

function $bar(context: DecoratorContext, target: Type) {
  context.call($foo, target);
}
```

```ts
function $foo(context: DecoratorContext, target: Type, name: string) {}

function $bar(context: DecoratorContext, target: Type, name: string) {
  context.call($foo, target, `bar.${name}`);
}
```

## Attributes

- [x] ‚úÖ Recommended
- [x] üîß Fixable
- [x] üí≠ Requires type information
